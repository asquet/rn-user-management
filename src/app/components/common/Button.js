import React from 'react';
import { Text, TouchableOpacity, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  buttonBody: {
    borderWidth: 1,
    borderColor: 'steelblue',
    borderRadius: 3,
    alignItems: 'center',
    padding: 5,
  },
  buttonText: {
    color: 'steelblue',
    fontSize: 20,
    fontWeight: 'bold',
  },
});

export default function Button({ buttonText, onPress, bodyStyles }) {
  return (
    <TouchableOpacity style={[styles.buttonBody, bodyStyles]} onPress={onPress}>
      <Text style={styles.buttonText}>
        {buttonText}
      </Text>
    </TouchableOpacity>
  );
}

Button.propTypes = {
  buttonText: React.PropTypes.string.isRequired,
  onPress: React.PropTypes.func.isRequired,
  bodyStyles: Text.propTypes.style,
};

Button.defaultProps = {
  bodyStyles: null,
};
