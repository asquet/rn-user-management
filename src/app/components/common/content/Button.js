import React from 'react';
import { Button as TheirButton } from 'react-native-elements';
import { formStyles } from '../../../stylesheet';

export default function Button({ buttonStyle, ...props }) {
  return (
    <TheirButton buttonStyle={[formStyles.buttonBody, buttonStyle]} {...props} />
  );
}

Button.propTypes = {
  ...TheirButton.propTypes,
};

Button.defaultProps = {
  ...TheirButton.defaultProps,
};
