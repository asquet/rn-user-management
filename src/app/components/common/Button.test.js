import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

// Note: test renderer must be required after react-native.

describe('Button', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Button buttonText="Button" onPress={jest.fn()} />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
