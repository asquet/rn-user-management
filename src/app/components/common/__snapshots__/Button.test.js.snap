// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "steelblue",
      "borderRadius": 3,
      "borderWidth": 1,
      "opacity": 1,
      "padding": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "steelblue",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Button
  </Text>
</View>
`;
